Objects/complexobject.c

```diff
@@ -156,14 +156,16 @@ _Py_c_pow(Py_complex a, Py_complex b)
     return r;
 }
 
+#define INT_EXP_CUTOFF 100
+
 static Py_complex
 c_powu(Py_complex x, long n)
 {
-    Py_complex r, p;
+    Py_complex p = x, r = n-- ? p : c_1;
     long mask = 1;
-    r = c_1;
-    p = x;
-    while (mask > 0 && n >= mask) {
+    assert(-1 <= n && n <= INT_EXP_CUTOFF);
```

Comment:
Don't you need `n <= INT_EXP_CUTOFF` to be checked before `n--`?

Reply:
Rather, I should use a strict inequality here.

Reply:
In general, we split `assert` so that we know which condition was actually violated. How about it?

Reply:
That does make sense.